packages/tabs/src/Tabs/Tabs.tsx

```diff
@@ -35,54 +48,54 @@ import { AccessibleTabsProps } from './Tabs.types';
  * @param props.children Content to appear inside of Tabs component.
- * @param props.setSelected Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
- * @param props.selected Index of the Tab that should appear active. If value passed, component will be controlled by consumer.
  * @param props.className className applied to Tabs container.
- * @param props.as HTML Element that wraps name in Tab List.
+ * @param props.selected Index of the Tab that should appear active. If value passed, component will be controlled by consumer.
+ * @param props.setSelected Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
  */
-function Tabs(props: AccessibleTabsProps) {
+const Tabs = (props: AccessibleTabsProps) => {
   validateAriaLabelProps(props, 'Tabs');
 
   const {
-    children,
-    inlineChildren,
-    className,
     as = 'button',
     baseFontSize: baseFontSizeProp,
-    setSelected: setControlledSelected,
-    selected: controlledSelected,
+    children,
+    className,
     darkMode: darkModeProp,
+    inlineChildren,
+    selected: controlledSelected,
+    setSelected: setControlledSelected,
     'data-lgid': dataLgId = LGIDS_TABS.root,
     'aria-labelledby': ariaLabelledby,
     'aria-label': ariaLabel,
     ...rest
   } = props;
+
   const baseFontSize: BaseFontSize = useUpdatedBaseFontSize(baseFontSizeProp);
   const { theme, darkMode } = useDarkMode(darkModeProp);
-
-  const [tabNode, setTabNode] = useState<HTMLDivElement | null>(null);
-  const [panelNode, setPanelNode] = useState<HTMLDivElement | null>(null);
-
   const accessibilityProps = {
     ['aria-label']: ariaLabel,
     ['aria-labelledby']: ariaLabelledby,
   };
 
+  const id = useIdAllocator({ prefix: rest.id || 'tabs' });
+  const { descendants: tabDescendants, dispatch: tabDispatch } =
+    useInitDescendants<HTMLDivElement>();
+  const { descendants: tabPanelDescendants, dispatch: tabPanelDispatch } =
+    useInitDescendants<HTMLDivElement>();
+
+  const [uncontrolledSelected, setUncontrolledSelected] = useState(0);
+  const [selected, setSelected] = [
+    controlledSelected ? controlledSelected : uncontrolledSelected,
+    controlledSelected ? setControlledSelected : setUncontrolledSelected,
+  ];
+
   const childrenArray = useMemo(
     () => React.Children.toArray(children) as Array<React.ReactElement>,
     [children],
   );
 
-  const isControlled = typeof controlledSelected === 'number';
-  const [uncontrolledSelected, setUncontrolledSelected] = useState(
-    childrenArray.findIndex(child => child.props.default || 0),
-  );
-  const selected = isControlled ? controlledSelected : uncontrolledSelected;
-  const setSelected = isControlled
-    ? setControlledSelected
-    : setUncontrolledSelected;
-
   const handleChange = useCallback(
     (e: React.SyntheticEvent<Element, MouseEvent>, index: number) => {
       setSelected?.(index);
```

Comment:
nit: I know this isn't your new code, but this handler name is super vague. Handle what change?